qabelbox/src/main/java/de/qabel/qabelbox/dagger/modules/ApplicationModule.java

```diff
@@ -48,6 +51,13 @@ ChatService providesChatService(IdentityRepository identityRepository, ContactRe
                 contactRepository, chatDropMessageRepository, dropStateRepository);
     }
 
+    @Singleton
+    @Provides
+    ChatServiceUseCase providesChatManager(ContactRepository contactRepo, ChatDropMessageRepository chatDropMessageRepository,
+                                           IdentityRepository identityRepo, ChatMessageTransformer msgTransformer){
+        return new MainChatServiceUseCase(chatDropMessageRepository, contactRepo, identityRepo, msgTransformer);
```

Comment:
Same here: Just use "MainChatServiceUseCase useCase" as te parameter to the provides method and Dagger will construct the object for you.